src/common/broadcast_strategy.hpp

```diff
@@ -42,8 +42,10 @@ enum class broadcasting_strategy_t {
 
 using bcast_set_t = std::set<broadcasting_strategy_t>;
 
-static const bcast_set_t default_strategies {broadcasting_strategy_t::scalar,
+inline const bcast_set_t default_strategies() {
+    return {broadcasting_strategy_t::scalar,
```

Comment:
Shouldn't it be `const bcast_set_t` object with static storage duration that is returned be a `const` reference to avoid creating a new `std::set` every time `default_strategies()` is called?

Reply:
Maybe. I'll make that change.